engine/common/api_helpers/filters.py

```diff
@@ -66,6 +67,13 @@ class MultipleChoiceCharFilter(filters.ModelMultipleChoiceFilter):
     pass
 
 
+@extend_schema_field(serializers.CharField)
+class ModelChoiceCharFilter(filters.ModelChoiceFilter):
```

Comment:
maybe call it `ModelChoicePublicPrimaryKeyFilter` so it's easier to understand what it's used for? can also add a default `to_field_name="public_primary_key"` in there